### ISSUE TITLE
WorkshopDL & Maps

### ISSUE THREAD

**Issue by peterkuzik** [Facepunch/garrysmod-issues#5428]
### Details
We are using WorkshopDL to provide maps and content to our players. This worked fine since forever. Map is added on the server trough hostworkshopcollection parameter and client is requested to download via. resource.AddWorkshop.
Since 2 weeks or longer all users on multiple branches (chormium and normal) experience an issue on certain maps. Only on maps that were added in the last few weeks. 

If we now take a recently added map as example:

Some users / everyone who did not subscribe to the collection gets the following errors on join:
example: rp_nigeria_jungle.bsp

2870615710 = rp_nigeria_jungle
```
WorkshopDL: Took too long to get item info for 2870615710!
WorkshopDL: There is no file to mount for '2870615710'!
Error downloading 'maps/rp_nigeria_jungle.bsp' (denied by server)
Error downloading 'maps/graphs/rp_nigeria_jungle.ain' (denied by server)
Missing map 'maps/rp_nigeria_jungle.bsp', disconnecting

```

at the same time, maps that we´re added long times ago, such as the venator map, work just fine - also for users that didnt subscribe the collection.

Since two years we were always using a combination of 3 collections, 2 of them added onto the big one too keep them organized. It dosnt change anything putting the affected maps into the main one.

**Comment by peterkuzik** [comment]
Workshop id for rp_nigeria_jungle is 2124858545 not 2870615710. But same error!

```
WorkshopDL: Took too long to get item info for 2124858545!
WorkshopDL: There is no file to mount for '2124858545'!
```

**Comment by Pdzly** [comment]
Did you try on different servers? or local dedicated server? is the map in the collection of the server? is the map on local in the /maps dir AND in the workshop?

**Comment by peterkuzik** [comment]
It is happening on all branches. Im affected with this issue aswell. Just reverted back from chromium to no beta and still having this issue.

Addons are added via. resource.AddWorkshop by Server, i have my content settings to allow all (clientside) and i am not subscribed to any addons currently.
Most addons load in, but a few dont. See this log:
`WorkshopDL: Not downloading/mounting addon 'Vanilla's Hyperspace Tool 2' (2106004687): Addon too new to use old API!
WorkshopDL: There is no file to mount for '2106004687'!
WorkshopDL: Took too long to get item info for 2542791276!
WorkshopDL: There is no file to mount for '2542791276'!
WorkshopDL: Took too long to get item info for 2555147366!
WorkshopDL: There is no file to mount for '2555147366'!
WorkshopDL: Took too long to get item info for 2569947212!
WorkshopDL: There is no file to mount for '2569947212'!
WorkshopDL: Took too long to get item info for 2576327840!
WorkshopDL: There is no file to mount for '2576327840'!
WorkshopDL: Took too long to get item info for 2605967172!
WorkshopDL: There is no file to mount for '2605967172'!
WorkshopDL: Took too long to get item info for 2660719701!
WorkshopDL: There is no file to mount for '2660719701'!
WorkshopDL: Took too long to get item info for 2670800372!
WorkshopDL: There is no file to mount for '2670800372'!
WorkshopDL: Took too long to get item info for 2735655223!
WorkshopDL: There is no file to mount for '2735655223'!
WorkshopDL: Took too long to get item info for 2795572550!
WorkshopDL: There is no file to mount for '2795572550'!
WorkshopDL: Took too long to get item info for 2798566826!
WorkshopDL: There is no file to mount for '2798566826'!
WorkshopDL: Took too long to get item info for 2841786055!
WorkshopDL: There is no file to mount for '2841786055'!
WorkshopDL: Took too long to get item info for 1810304039!
WorkshopDL: There is no file to mount for '1810304039'!
WorkshopDL: Took too long to get item info for 766599047!
WorkshopDL: There is no file to mount for '766599047'!
WorkshopDL: Took too long to get item info for 2873823992!
WorkshopDL: There is no file to mount for '2873823992'!
WorkshopDL: Took too long to get item info for 2873768237!
WorkshopDL: There is no file to mount for '2873768237'!
WorkshopDL: Took too long to get item info for 2882315209!
WorkshopDL: There is no file to mount for '2882315209'!
WorkshopDL: Took too long to get item info for 2898552692!
WorkshopDL: There is no file to mount for '2898552692'!
WorkshopDL: Took too long to get item info for 2899162749!
WorkshopDL: There is no file to mount for '2899162749'!
WorkshopDL: Took too long to get item info for 2902246709!
WorkshopDL: There is no file to mount for '2902246709'!
WorkshopDL: Took too long to get item info for 1133172418!
WorkshopDL: There is no file to mount for '1133172418'!
WorkshopDL: Took too long to get item info for 2144176587!
WorkshopDL: There is no file to mount for '2144176587'!
WorkshopDL: Took too long to get item info for 2921479434!
WorkshopDL: There is no file to mount for '2921479434'!
WorkshopDL: Took too long to get item info for 2929627396!
WorkshopDL: There is no file to mount for '2929627396'!
WorkshopDL: Took too long to get item info for 1563313391!
WorkshopDL: There is no file to mount for '1563313391'!
WorkshopDL: Took too long to get item info for 2009233080!
WorkshopDL: There is no file to mount for '2009233080'!
WorkshopDL: Took too long to get item info for 2882815461!
WorkshopDL: There is no file to mount for '2882815461'!
WorkshopDL: Took too long to get item info for 2505065850!
WorkshopDL: There is no file to mount for '2505065850'!
WorkshopDL: Took too long to get item info for 112264602!
WorkshopDL: There is no file to mount for '112264602'!
WorkshopDL: Took too long to get item info for 2819288256!
WorkshopDL: There is no file to mount for '2819288256'!
WorkshopDL: Took too long to get item info for 144936535!
WorkshopDL: There is no file to mount for '144936535'!
WorkshopDL: Took too long to get item info for 2095413996!
WorkshopDL: There is no file to mount for '2095413996'!
WorkshopDL: Took too long to get item info for 2101847480!
WorkshopDL: There is no file to mount for '2101847480'!
WorkshopDL: Took too long to get item info for 2034593720!
WorkshopDL: There is no file to mount for '2034593720'!
WorkshopDL: Took too long to get item info for 1843119243!
WorkshopDL: There is no file to mount for '1843119243'!
`

ive checked steam workshopdl logs.. nothing that unusual i guess.
[workshop_log.txt](https://github.com/Facepunch/garrysmod-issues/files/11068915/workshop_log.txt)

**Comment by peterkuzik** [comment]
It also works sometimes. I really think is steam thats broken somehow. Did experience slight errors with steam workshop webpages lately.

**Comment by robotboy655** [comment]
Yeah I have been getting random server errors from Steam as well lately. Perhaps it is just their servers having issues.

**Comment by MrPupir** [comment]
+1, same error
download stucks after 102 addon :/
no matter which addon, if you swap their priorities, the error still remains

**Comment by oninoni** [comment]
Got the same issue right now.

**Comment by oninoni** [comment]
Seems like steam workshop is enforcing a rate limit on "100 Addons per user requesting addons on a specific server"
We have 2 identical servers. 1 Works. Other one does not.

**Comment by harland555** [comment]
This issue has returned yes correct, the limit is 102 addons, then it will timeout.

**Comment by oninoni** [comment]
The issue seems to be mitigated on dev branch due to a fix mentioned in #5409

**Comment by CryptAlchemy** [comment]
Bump.

**Comment by robotboy655** [comment]
Duplicate of https://github.com/Facepunch/garrysmod-issues/issues/5409 as far as I can tell. Already fixed on `dev` beta (clientside).
